sfge/src/test/java/com/salesforce/rules/multiplemassschemalookup/MultipleMassSchemaLookupRuleTest.java

```diff
@@ -0,0 +1,297 @@
+                "   }\n" +
+                "}\n";
+        // spotless:on
+
+        assertViolations(
+                RULE,
+                sourceCode,
+                expect(
+                        9,
+                        RuleConstants.METHOD_SCHEMA_GET_GLOBAL_DESCRIBE,
+                        4,
+                        "MyClass",
+                        RuleConstants.RepetitionType.LOOP,
+                        loopAstLabel));
+    }
+
+    @CsvSource({
+        "ForEachStatement, for (String s1 : myList), for (String s2 : myList)",
+        "ForLoopStatement, for (Integer i; i < myList.size; s++), for (Integer j; j < myList.size; s++)",
+        "WhileLoopStatement, while(true), while(true)"
+    })
+    @ParameterizedTest(name = "{displayName}: {0}")
+    public void testLoopWithinLoop(
+            String loopAstLabel, String loopStructure1, String loopStructure2) {
+        // spotless:off
+        String sourceCode =
+                "public class MyClass {\n" +
+                "   void foo(String[] myList) {\n" +
+                    "   " + loopStructure1 + " {\n" +
+                    "       " + loopStructure2 + " {\n" +
+                    "           Schema.getGlobalDescribe();\n" +
+                    "       }\n" +
+                    "   }\n" +
+                    "}\n" +
+                "}\n";
+        // spotless:on
+
+        assertViolations(
+                RULE,
+                sourceCode,
+                expect(
+                        5,
+                        RuleConstants.METHOD_SCHEMA_GET_GLOBAL_DESCRIBE,
+                        3,
+                        "MyClass",
+                        RuleConstants.RepetitionType.LOOP,
+                        loopAstLabel),
+                expect(
+                        5,
+                        RuleConstants.METHOD_SCHEMA_GET_GLOBAL_DESCRIBE,
+                        4,
+                        "MyClass",
+                        RuleConstants.RepetitionType.LOOP,
+                        loopAstLabel));
+    }
+
+    @CsvSource({
+        "GGD followed by GGD, Schema.getGlobalDescribe, Schema.getGlobalDescribe",
+        "GGD followed by DSO, Schema.getGlobalDescribe, Schema.describeSObjects",
+        "DSO followed by DSO, Schema.describeSObjects, Schema.describeSObjects",
+        "DSO followed by GGD, Schema.describeSObjects, Schema.getGlobalDescribe"
+    })
+    @ParameterizedTest(name = "{displayName}: {0}")
+    public void testMultipleInvocations(String testName, String firstCall, String secondCall) {
+        String sourceCode =
+                "public class MyClass {\n"
+                        + "   void foo() {\n"
+                        + "       "
+                        + firstCall
+                        + "();\n"
+                        + "       "
+                        + secondCall
+                        + "();\n"
+                        + "   }\n"
+                        + "}\n";
```

Comment:
For the future, you can use `spotless:off` and `spotless:on` to avoid lines being split like this. Keeps the code readable.

Reply:
Agree. This got missed. Will clean it up.